script.js

```diff
@@ -157,43 +324,293 @@ class Game {
     }
   }
 
-  createApple() {
+  createObstacle() {
     let x = 0;
     let y = 0;
     
+    let valid = [];
```

Comment:
You don't need that array here. What you want to do in the loop below is:
1. Set `isValid` to `true`
2. Validate all things (snake, apples, obstacles) - if any of them collides with you `(x, y)` coordinates, set `isValid` to `false`. Otherwise, don't change it.
3. If at the end of the loop `isValid` is still `true`, that means there were no collisions, therefore the position is valid.